server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -2015,16 +2080,24 @@ public void testSupportsParallelCollectionAffectsSlicing() throws Exception {
         }
     }
 
-    public void testSupportsParallelCollection() {
+    public void testIsParallelCollectionSupportedForResults() {
         SearchSourceBuilder searchSourceBuilder = randomBoolean() ? null : new SearchSourceBuilder();
         if (searchSourceBuilder != null && randomBoolean()) {
             searchSourceBuilder.aggregation(new TermsAggregationBuilder("terms"));
         }
-        assertTrue(SearchService.supportsParallelCollection(ResultsType.DFS, searchSourceBuilder));
+        assertTrue(SearchService.isParallelCollectionSupportedForResults(ResultsType.DFS, searchSourceBuilder, true));
+        assertFalse(
+            SearchService.isParallelCollectionSupportedForResults(
+                randomFrom(randomFrom(ResultsType.QUERY, ResultsType.NONE, ResultsType.FETCH)),
+                searchSourceBuilder,
+                false
+            )
+        );
         assertFalse(
```

Comment:
This assertFalse statement seems the same as the previous, but expects `true`. What gives?? (doesn't make sense to me - and I see this test fail intermittently)

Reply:
I filed #98534 to track this.